### ISSUE TITLE
All COSMIC app windows should have a small neutral border and drop shadows

### ISSUE THREAD

**Issue by maria-komarova** [pop-os/libcosmic#421]
COSMIC app windows should have a small 1px border (@bg_divider color) inside each window area and a drop shadow: @shade_color which is defined as rgba (0, 0, 0, 0.32) for Dark mode, and rgba (0, 0, 0, 0.08) for Light mode.

**Comment by TheESN** [comment]
> This is not a cosmic-comp issue, as it doesn't draw these types of decorations, instead the apps do. This is a libcosmic issue.

What about applications with server side decorations?

**Comment by maria-komarova** [comment]
I'd include them as well unless they have some other styling.

**Comment by git-f0x** [comment]
Would this also apply to dialogs, modals and maybe applet popups?

**Comment by maria-komarova** [comment]
Ah, right, it should - dialogs, all menu types and stand-alone notifications. I wonder if we need a separate issue for this.

**Comment by mmstick** [comment]
@PoisonFrog It's not technically feasible to have precise styling across toolkits, as that requires fundamentally altering how those toolkits theme and render their widgets. We can only provide a best effort based on the theming capabilities those toolkits make configurable to the OS.

**Comment by PoisonFrog** [comment]
> @PoisonFrog It's not technically feasible to have precise styling across toolkits, as that requires fundamentally altering how those toolkits theme and render their widgets. We can only provide a best effort based on the theming capabilities those toolkits make configurable to the OS.

I understand, but I was talking about borders and shadows only. Borders and shadows are quite consistent across toolkits in KDE and XFCE. I understand that the look and feel of the apps in general will not be consistent across toolkits.

**Comment by TheESN** [comment]
> > @PoisonFrog It's not technically feasible to have precise styling across toolkits, as that requires fundamentally altering how those toolkits theme and render their widgets. We can only provide a best effort based on the theming capabilities those toolkits make configurable to the OS.
> 
> I understand, but I was talking about borders and shadows only. Borders and shadows are quite consistent across toolkits in KDE and XFCE. I understand that the look and feel of the apps in general will not be consistent across toolkits.

This consistency is achieved by usage of Server Side Decorations. I think that applications that utilizes them will look consistent in COSMIC

**Comment by edfloreshz** [comment]
Window borders were implemented in #679! 🎉

**Comment by PoisonFrog** [comment]
libcosmic windows look better now, but I can't help but notice that the window borders are different than the libadwaita ones which is understandable since it is a different toolkit that also uses client side decorations, but the libadwaita borders look "smoother" to me. Maybe it is because libcosmic apps do not project shadows yet idk. Anyone else feels the same when comparing libcosmic and libadwaita window borders or is it just me?

**Comment by ryanabx** [comment]
Shadows are probably blocked on https://github.com/rust-windowing/winit/issues/3438 , which once implemented will need to be propagated up to iced, and then libcosmic
